### DISCUSSION TITLE
Server Sent Event for Plugins

### DISCUSSION THREAD

**Discussion by 2547techno** [Chatterino/chatterino2#5526] [Ideas] [answered]
With the recent addition of network capabilities for plugins, real-time events would also be nice.

My thought was that SSE would be a lot easier to implement rather than WebSockets, but WebSockets could still be a future goal.

The interface being something like: `HTTPRequest:on_message(callback)` and maybe an `HTTPRequest:on_close(callback)` if its needed

**Comment by Mm2PL** [comment]
WebSockets are just plain better and we already have them (in Chatterino not in the plugins API)

**Reply by 2547techno** [reply]
I see, are there already plans to introduce it to plugins that I missed?

**Reply by Mm2PL** [reply, marked as answer]
I would want WS for plugins however it's quite a bit of work. I also need to fix the asynchronous crashes I am dealing with.

**Reply by 2547techno** [reply]
ah ok, that's fair
